Packages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthTransactionHelper.cs

```diff
@@ -0,0 +1,56 @@
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace ChainSafe.GamingSdk.Web3Auth
+{
+    /// <summary>
+    /// Transaction helpers class to halt transaction processing when the Web3Auth wallet GUI is active.
+    /// </summary>
+    public static class Web3AuthTransactionHelper
+    {
+
+        #region Properties
+
+        public static TaskCompletionSource<bool> TransactionAcceptedTcs { get; set; }
+
+        public static bool Working { get; set; }
+
+        #endregion
+
+        #region Methods
+        
+        /// <summary>
+        /// Waits for transcation confirmation.
+        /// </summary>
+        public static async Task WaitForTransactionAsync()
+        {
+            Debug.Log("Waiting for Web3AuthWallet tx confirmation");
+            TransactionAcceptedTcs = new TaskCompletionSource<bool>();
+            Working = true;
+            await TransactionAcceptedTcs.Task;
+        }
+        
+        /// <summary>
+        /// Accepts a pending transcation.
+        /// </summary>
+        public static void AcceptTransaction()
+        {
+            Debug.Log("Accepting Web3AuthWallet transaction");
+            TransactionAcceptedTcs.SetResult(true);
+            Working = false;
+            TransactionAcceptedTcs = null;
```

Comment:
because there's a polling existence check in the tx manager for the tcs for incoming transactions before it has completed, didn't want it to loop so i put hard stop there for safety.

![image](https://github.com/ChainSafe/web3.unity/assets/57473220/8b8f1298-976c-48a7-9bbe-d7089471c632)

Reply:
You really shouldn't poll for incoming transactions, you can subscribe to an event that's invoked in txn executor when a transaction is requested